### PR TITLE
Add middleware

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/backend 
@maxpeterson 

Copied middleware function as its been used in 2 other projects

### CODE REVIEW

File: declaration/templates/declaration/_declaration_form.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<div class="declaration-form">
+    <h1>Are you an HCP?</h1>
+    <div class="bag-image"></div>
```
Comment (maxpeterson) [inline declaration/templates/declaration/_declaration_form.html]: We should remove this `bag-image`.

File: declaration/middleware.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from django.http import HttpResponseRedirect
+from django.template.defaultfilters import urlencode
+
+from .views import DECLARATION_COOKIE_KWARGS
+
+
+PROTECTED_URL_PATTERNS = settings.PROTECTED_URL_PATTERNS
+
+
+class DeclarationMiddleware(object):
```
Comment (todd-dembrey) [inline declaration/middleware.py:14]: Do we want to be compatible with 1.10 and greater?

https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware

File: declaration/templates/declaration/_declaration_form.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    <p><strong>I am</strong> a Healthcare professional</p>
+    <form action="{% url 'declaration:form' %}" method="post">
+        {% csrf_token %}
+        {{ form }}
+        <button type="submit">Yes</button>
```
Comment (maxpeterson) [inline declaration/templates/declaration/_declaration_form.html]: Looking at the rest of the wording `Yes` seems wrong and for completeness it should probably have a `name` and `value`:

```
<button type="submit" name="declaration" value="yes">I confirm I am a HCP</button>
```

**Comment by maxpeterson** [comment]
Looks good.

File: declaration/middleware.py (hunk 2 above)
Comment (maxpeterson) [reply declaration/middleware.py:14]: I would like it to be compatible although it is not essential now. Is `django.utils.deprecation.MiddlewareMixin` available in earlier versions of Django?

Comment (todd-dembrey) [reply declaration/middleware.py:14]: No. This is the pattern I have seen being used.

```
try:
    from django.utils.deprecation import MiddlewareMixin
except ImportError:
    MiddlewareMixin = object
```

Comment (maxpeterson) [reply declaration/middleware.py:14]: Let's look at adding this in the future, adding it now will make the library incompatible with earlier versions of Django.

Comment (todd-dembrey) [reply declaration/middleware.py:14]: That wont happen using the import statement above. But if it isnt needed for now thats fine.

**Pull request merged**